src/frontend/components/app/action-header/styled-back-button.tsx

```diff
@@ -1,41 +1,33 @@
 import React from 'react'
 import styled from 'styled-components'
-import { Link as RouterLink } from 'react-router-dom'
+import { useNavigation, Link as RouterLink, useNavigate } from 'react-router-dom'
 import {
   ButtonCSS,
   ButtonProps,
   Icon,
 } from '@adminjs/design-system'
 
-import { useSelector } from 'react-redux'
 import ViewHelpers from '../../../../backend/utils/view-helpers/view-helpers'
-import { ReduxState, RouterProps } from '../../../store'
 import allowOverride from '../../../hoc/allow-override'
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const StyledLink = styled(({ rounded, ...rest }) => <RouterLink {...rest} />)<ButtonProps>`${ButtonCSS}`
 const h = new ViewHelpers()
 
 export type StyledBackButtonProps = {
-  resourceId: string;
   showInDrawer: boolean;
 }
 
 const StyledBackButton: React.FC<StyledBackButtonProps> = (props) => {
-  const { resourceId, showInDrawer } = props
+  const { showInDrawer } = props
   const cssCloseIcon = showInDrawer ? 'ChevronRight' : 'ChevronLeft'
 
-  const { from } = useSelector<ReduxState, RouterProps>((state) => state.router)
-  const { pathname, search } = from
-
-  const backButtonUrl = pathname
-    ? [pathname, search].join('')
-    : h.resourceUrl({ resourceId, search: window.location.search })
+  const navigate = useNavigate()
 
   return (
     <StyledLink
       size="icon"
-      to={backButtonUrl}
+      to={navigate(-1)}
```

Comment:
How does it work currently? Didn't we decide to remove it completely from non-drawer ui?

Reply:
fixed